sdks/python/apache_beam/examples/inference/sklearn_japanese_housing_regression.py

```diff
@@ -0,0 +1,164 @@
+https://www.kaggle.com/datasets/nishiodens/japan-real-estate-transaction-prices
+
+Since the data has missing fields, this example illustrates how to split
+data and assign it to the models that are trained on different subsets of
+features. The predictions are then recombined.
+
+In order to set this example up, you will need two things.
+1. Build models (or use ours) and reference those via the model directory.
+2. Download the data from kaggle and host it.
+"""
+
+import argparse
+from typing import Iterable
+
+import apache_beam as beam
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.ml.inference.sklearn_inference import ModelFileType
+from apache_beam.ml.inference.sklearn_inference import SklearnModelHandlerPandas
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+import pandas
+
+MODELS = [{
+    'name': 'all_features',
+    'required_features': [
+        'Area',
+        'Year',
+        'MinTimeToNearestStation',
+        'MaxTimeToNearestStation',
+        'TotalFloorArea',
+        'Frontage',
+        'Breadth',
+        'BuildingYear'
+    ]
+},
+          {
+              'name': 'floor_area',
+              'required_features': ['Area', 'Year', 'TotalFloorArea']
+          },
+          {
+              'name': 'stations',
+              'required_features': [
+                  'Area',
+                  'Year',
+                  'MinTimeToNearestStation',
+                  'MaxTimeToNearestStation'
+              ]
+          }, {
+              'name': 'no_features', 'required_features': ['Area', 'Year']
+          }]
+
+
+def sort_by_features(dataframe, max_size):
+  """ Partitions the dataframe by what data it has available."""
+  for i, model in enumerate(MODELS):
+    required_features = dataframe[model['required_features']]
+    if not required_features.isnull().any().any():
+      return i
+  return -1
+
+
+class LoadDataframe(beam.DoFn):
+  def process(self, file_name: str) -> Iterable[pandas.DataFrame]:
+    """ Loads data files as a pandas dataframe."""
+    file = FileSystems.open(file_name, 'rb')
+    dataframe = pandas.read_csv(file)
+    for i in range(dataframe.shape[0]):
+      yield dataframe.iloc[[i]]
+
+
+def report_predictions(prediction_result):
+  true_result = prediction_result.example['TradePrice'].values[0]
+  inference = prediction_result.inference
+  return 'True Price %.1f, Predicted Price %f' % (true_result, inference)
+
+
+def parse_known_args(argv):
+  """Parses args for the workflow."""
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--input',
+      dest='input',
+      required=True,
+      help='A metadata file with all models, with references to models and '
+      'information about all the files and data.')
+  parser.add_argument(
+      '--model_path',
+      dest='model_path',
```

Comment:
Hrm, I prefer model_path.  I think even though there are many models each model will have the same path. I was trying to think of why, so if you had a path containing libraries, you would probably call it lib_path or library_path rather than libraries_path.